`pip install mercurial` is broken on Windows CI

https://github.com/crystal-lang/shards/actions/runs/10661704235/job/29547633885

```
Run pip install mercurial
  pip install mercurial
  shell: C:\Program Files\PowerShell\7\pwsh.EXE -command ". '{0}'"
Collecting mercurial
  Downloading mercurial-6.8.1.tar.gz (8.3 MB)
     ---------------------------------------- 8.3/8.3 MB 74.0 MB/s eta 0:00:00
  Installing build dependencies: started
  Installing build dependencies: finished with status 'done'
  Getting requirements to build wheel: started
  Getting requirements to build wheel: finished with status 'error'
  error: subprocess-exited-with-error
  
  Getting requirements to build wheel did not run successfully.
  exit code: 1
  
  [17 lines of output]
  Traceback (most recent call last):
    File "C:\hostedtoolcache\windows\Python\3.9.13\x64\lib\site-packages\pip\_vendor\pyproject_hooks\_in_process\_in_process.py", line 353, in <module>
      main()
    File "C:\hostedtoolcache\windows\Python\3.9.13\x64\lib\site-packages\pip\_vendor\pyproject_hooks\_in_process\_in_process.py", line 335, in main
      json_out['return_val'] = hook(**hook_input['kwargs'])
    File "C:\hostedtoolcache\windows\Python\3.9.13\x64\lib\site-packages\pip\_vendor\pyproject_hooks\_in_process\_in_process.py", line 118, in get_requires_for_build_wheel
      return hook(config_settings)
    File "C:\Users\runneradmin\AppData\Local\Temp\pip-build-env-f47lhb64\overlay\Lib\site-packages\setuptools\build_meta.py", line 332, in get_requires_for_build_wheel
      return self._get_build_requires(config_settings, requirements=[])
    File "C:\Users\runneradmin\AppData\Local\Temp\pip-build-env-f47lhb64\overlay\Lib\site-packages\setuptools\build_meta.py", line 302, in _get_build_requires
      self.run_setup()
    File "C:\Users\runneradmin\AppData\Local\Temp\pip-build-env-f47lhb64\overlay\Lib\site-packages\setuptools\build_meta.py", line 503, in run_setup
      super().run_setup(setup_script=setup_script)
    File "C:\Users\runneradmin\AppData\Local\Temp\pip-build-env-f47lhb64\overlay\Lib\site-packages\setuptools\build_meta.py", line 318, in run_setup
      exec(code, locals())
    File "<string>", line 1665, in <module>
  ImportError: cannot import name 'msvccompiler' from 'distutils' (C:\Users\runneradmin\AppData\Local\Temp\pip-build-env-f47lhb64\overlay\Lib\site-packages\setuptools\_distutils\__init__.py)
  [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: subprocess-exited-with-error

Getting requirements to build wheel did not run successfully.
exit code: 1

See above for output.

note: This error originates from a subprocess, and is likely not a problem with pip.
Error: Process completed with exit code 1.
```

The error reproduces with `pip install mercurial` on my local machine, so it's not specific to the CI environment.

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -36,6 +36,7 @@ jobs:
 
       - name: Install Mercurial
         run: pip install mercurial
+        if: ${{ matrix.os != 'windows-2022' }}
 
       - name: Install Crystal
         uses: oprypin/install-crystal@v1
@@ -54,9 +55,9 @@ jobs:
           brew update
           brew install fossil
 
-      - name: Install Fossil
+      - name: Install Fossil and Mercurial
         if: ${{ matrix.os == 'windows-2022' }}
-        run: choco install fossil
+        run: choco install fossil hg
 
       - name: Download source
         uses: actions/checkout@v4

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -37,13 +37,13 @@ jobs:
         if: ${{ runner.os == 'Linux' }}
         run: |
           sudo apt-get update
-          sudo apt-get install fossil hg
+          sudo apt-get install fossil mercurial
 
       - name: Install Fossil and Mercurial
         if: ${{ runner.os == 'macOS' }}
         run: |
           brew update
-          brew install fossil hg
+          brew install fossil mercurial
 
       - name: Install Fossil and Mercurial
         if: ${{ runner.os == 'Windows' }}

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -28,35 +28,25 @@ jobs:
           git config --global column.ui always
           git config --global core.autocrlf false
 
-      - name: Install Python
-        uses: actions/setup-python@v5
-
-      - name: Upgrade pip
-        run: python -m pip install --upgrade pip
-
-      - name: Install Mercurial
-        run: pip install mercurial
-        if: ${{ matrix.os != 'windows-2022' }}
-
       - name: Install Crystal
         uses: oprypin/install-crystal@v1
         with:
           crystal: ${{ matrix.crystal }}
 
-      - name: Install Fossil
+      - name: Install Fossil and Mercurial
         if: ${{ runner.os == 'Linux' }}
         run: |
           sudo apt-get update
-          sudo apt-get install fossil
-
-      - name: Install Fossil
-        if: ${{ matrix.os == 'macos-12' }}
-        run: |
-          brew update
-          brew install fossil
+          sudo apt-get install fossil hg
 
       - name: Install Fossil and Mercurial
-        if: ${{ matrix.os == 'windows-2022' }}
+        if: ${{ runner.os == 'macOS' }}
+        run: |
+          brew update
+          brew install fossil hg
+
+      - name: Install Fossil and Mercurial
+        if: ${{ runner.os == 'Windows' }}
         run: choco install fossil hg
 
       - name: Download source

